core/prelude/types/uint.carbon

```diff
@@ -50,27 +50,27 @@ impl forall [N:! IntLiteral()] UInt(N) as Ordered {
 // Arithmetic.
 
 impl forall [N:! IntLiteral()] UInt(N) as Add {
-  fn Op[self: Self](other: Self) -> Self = "int.sadd";
+  fn Op[self: Self](other: Self) -> Self = "int.uadd";
 }
 
 impl forall [N:! IntLiteral()] UInt(N) as Div {
-  fn Op[self: Self](other: Self) -> Self = "int.sdiv";
+  fn Op[self: Self](other: Self) -> Self = "int.udiv";
 }
 
 impl forall [N:! IntLiteral()] UInt(N) as Mod {
-  fn Op[self: Self](other: Self) -> Self = "int.smod";
+  fn Op[self: Self](other: Self) -> Self = "int.umod";
 }
 
 impl forall [N:! IntLiteral()] UInt(N) as Mul {
-  fn Op[self: Self](other: Self) -> Self = "int.smul";
+  fn Op[self: Self](other: Self) -> Self = "int.umul";
 }
 
 impl forall [N:! IntLiteral()] UInt(N) as Negate {
   fn Op[self: Self]() -> Self = "int.snegate";
```

Comment:
```suggestion
  fn Op[self: Self]() -> Self = "int.unegate";
```
`snegate` traps on signed overflow (eg, `(-0x8000_0000 as i32) / -1` is an error); `unegate` does not. This would be visible in the overflow flags in lowering if we emitted them; for now I think it is only visible in constant evaluation.

Reply:
Fixed in #4744. (It turns out that `snegate` actually performed a `unegate` when used on an unsigned operand, so this was already functionally correct.)